### PR TITLE
athletics.lic add support for more efficient athletics learning #4062 

### PR DESCRIPTION
**Pull request by vtcifer**
Updated athletics.lic and base-athletics.yaml for climbing over and back prior to starting climb practice. https://github.com/rpherbig/dr-scripts/issues/4062

### CODE REVIEW

**Comment by Tarhish** [comment]
I love the love for Ratha, and this looks reasonable.  Looking at this, though, makes me think we could probably just add the extra lines to climb_practice and skip adding a new climb_practice2, like up above you'd call

```
climb_practice(@climbing_target['id'], @climbing_target['target'], @climbing_target['target2'], @climbing_target['hide'])
```

And then you'd add the argument to climb_practice, ` def climb_practice(room, target, target2, to_hide = false)` so that down below it would have:
```
move("climb #{target}") if target2
move("climb #{target2}") if target2
```

If there is a target2 then it'll get sent along to climb_practice, but if not then it'll remain nil so it won't do the moves.  Do you think that'd work?

I'll gladly try this out though.

**Comment by vtcifer** [comment]
Not a ruby expert by any stretch, so I'm not 100% on what works/what doesn't.  Most of what I'll be contributing will be filtered through learning from stackoverflow. My original thought on this one was to preserve backwards compatibility the only way I knew.  I can give this a shot and test.

**Comment by Tarhish** [comment]
If it's a problem we can look at condensing it later, too. I'd be willing to help with that.

**Comment by vtcifer** [comment]
Tested OK.  
I will add the strong caveat that the target2 option should only be added when you can make the climb both directions, so careful updates to base-athletics.yaml will be required to make use of this.

File: athletics.lic [hunk 1]
```diff
@@ -290,6 +302,8 @@ class Athletics
         start_script('performance', ['noclean']) unless Script.running?('performance')
         pause @performance_pause # make sure you are playing before starting climb practice
       end
+      move("climb #{target}") if target2
```
Comment (jayRyan24) [inline athletics.lic]: Possibly make this an array and just `move climb` for each if this is the best solution.

File: athletics.lic [hunk 2]
```diff
@@ -47,7 +47,7 @@ class Athletics
     elsif @swimming_target
       swim_loop(@swimming_target['rooms'])
     elsif @climbing_target
-      climb_practice(@climbing_target['id'], @climbing_target['target'], @climbing_target['hide'])
+      climb_practice(@climbing_target['id'], @climbing_target['target'], @climbing_target['target2'], @climbing_target['hide'])
```
Comment (jayRyan24) [inline athletics.lic]: If anything here, I'd consider making `target` an array and then doing the above for each of them. A good thing to consider is why this is being done. Is it stopping too early I take it? So, it climbs for 15 ranks but not 30? Something like that?

**Comment by jayRyan24** [comment]
Thanks for contributing to the system! It really does still need a lot of work! 

For this review, overall this is very kludge, and maybe not the best solution? I guess the main question is what is the issue? And then we should decide on the best way of fixing it, not just doing things multiple times within the code.

**Comment by vtcifer** [comment]
> I guess the main question is what is the issue?

This is the issue:
```
[Script finished (In 720.0249101 seconds): climb.cmd(19)]
[Script finished (In 722.0445302 seconds): climb.cmd(19)]
[Script finished (In 579.6748853 seconds): climb2.cmd(106)]
```

Climb JUST does climb practice. (I ran it twice, on each side of a climb to validate there wasn't a difference depending on direction).
Climb2 climbs over and back, then does climb practice until finished, then repeats. 
Climbing over an obstacle is essentially instant, and provides ~ 1/8 of the bits of a full climb practice, in near 0 time.  Climbing over and back provides about ~1/4 the bits of a full climb practice in near 0 time. 
By doing this, you gain an approx 20% efficiency. 

>And then we should decide on the best way of fixing it, not just doing things multiple times within the code.

I'm not sure what is being talked about here.  The double move? The tracking of multiple objects (not all climbs use the same object for back and forth)?

As I pointed out in my original reply, not a ruby expert by any stretch so more than happy to know what would work better with ruby.

**Pull request closed without merging**